_posts/2021-12-15-dd_00004_logic_gates_2.md

```diff
@@ -0,0 +1,369 @@
+</figure>
+
+As you can see in the picture, the upper parts are each connected to something
+called "5V". What does that mean? "V" stands for **Volts**, and as you probably
+guessed, it is the unit for **voltage**. In digital circuits, voltages are used
+to represent binary (or _logic_) 1s and 0s. Two voltages are sufficient, and
+usually the higher of the two is used to represent a binary 1, and the lower for
+a binary 0. Common voltages for binary 1s are 5V or 3.3V. In our example above,
+the drain (of the NMOS) is connected to 5 Volts. So this circuit uses 5V to
+represent a binary 1. What about the triangle thingies at the bottom? This is
+what we call **ground**. It is the _reference potential_[^1], and can be viewed
+effectively as 0V. You probably guessed it: 0V represents a binary 0.
+
+[^1]:
+    There are no _absolute_ voltages. A voltage is always a _potential
+    difference_. Therefore, every electric circuit needs some kind of reference
+    potential, a baseline, and all other voltages are relative to this
+    reference. We might do a future post on the basics of electronics where we
+    explain in depth what voltage, current etc are.
+
+In our post about [transistors](categories/digital_design/transistors), we told
+you that an NMOS turns on (i.e. opens its channel) when some voltage larger than
+its _threshold_ voltage is applied to its gate. The threshold voltage is
+different from transistor to transistor. For the NMOS transistors used in post,
+we assume that 5 Volts are well above their threshold voltage, and therefore,
+they turn on. The PMOS transistors behave opposite to NMOS transistors: they
+turn on when the gate voltage is 0V, and they turn off when 5V are applied to
+their gate. From now on, we will use the binary values for the gate voltages: 1
+means that 5V are applied to the gate, and 0 means 0V.
+
+Most modern chips make use of a technology called **CMOS**. CMOS stands for
+_Complementary Metal-oxide semiconductor_. The "MOS" part might be already
+familiar to you: this refers to the MOSFETs, we already learned about. The
+"complementary" means, that in CMOS circuits, always _pairs_ of NMOS and PMOS
+transistors are used. The reasons why this is done like this are beyond the
+scope of this post, but just keep that fact in mind. In the following we will
+see plenty of examples.
+
+## Inverter
+
+The most basic CMOS circuit is the **CMOS inverter**, and consists of two
+transistors. One PMOS and one NMOS:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00004-cmosinv.jpg" alt="CMOS inverter">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+The input on the left goes into the gates of both transistors. The output on the
+right is connected to the middle section between the two transistors. You may
+remember from our last post that an inverter just flips a 1 to a 0, and vice
```

Comment:
maybe at a link to last post

Reply:
I already have a link to it in the beginning of this post. I would say its not necessary to add another link?